### ISSUE TITLE
#Content: Scientific Domains Tab 

### ISSUE THREAD

**Issue by shaloo** [numpy/numpy.org#92]
Add content related to Scientific Domains in NumPy landing 'ecosystem' .

**Comment by shaloo** [comment]
@rgommers - here is the workflow I plan to follow for this content addition.

A. Make updates in shaloo/numpy.org fork -> branch issu92_sc_dom

B. Before Pull request
       git fetch upstream
       git checkout master
       git merge upstream/master

C.  Rebase local dev branch (issue92_sc_dom)
      git checkout issu92_sc_dom
      git rebase master

D.  Optional (if squashing is needed)
      git checkout
      git rebase -i master
E. Make a pull request from shaloo/numpy.org branch issu92_sc_dom to NumPy/numpy.org 'master'
      Go to GitHub in browser 
      Select issu92_sc_dom
      Compare with NumPy/numpy.org master - make sure all my updates show up
      Issue PR

Once PR is merged, this is what I typically do - may be I need to change this workflow to avoid sync up with master conflicts.

1. Sync up with upstream master
    git fetch upstream
    git checkout master
    git merge upstream/master
    git push origin master  (send updates to my fork master - is this the cause of merge issues that I am seeing?)
    
2. Validate that all issu92_sc_dom branch changes came back in resent with upstream master
     Github browser - compare issu92_sc_dom from my fork with upstream master and if all changes made it delete my local branch using this command
 git branch -d issu92_sc_dom

At times, I see errors indicating all changes in issu92_sc_dom are not present in upstream master though a diff does not show any changes.  For Data Science branch - i used a -D option and deleted it anyways .

3.  Create a new branch corresponding to new issue/ticket and add content for a new tab / new page on new website.

Repeat A-E and 1-2 for subsequent new feature/content addition.

**Comment by rgommers** [comment]
> @rgommers - here is the workflow I plan to follow for this content addition.
> 
> A. Make updates in shaloo/numpy.org fork -> branch issu92_sc_dom

Yes. If you make sure your `master` branch is up to date with `upstream/master` and that you create `issu92_sc_dom` from master, that will remove the need for much of what follows.

> 
> B. Before Pull request
> git fetch upstream
> git checkout master
> git merge upstream/master
> 
> C. Rebase local dev branch (issue92_sc_dom)
> git checkout issu92_sc_dom
> git rebase master
> 
> D. Optional (if squashing is needed)
> git checkout
> git rebase -i master
> 
> E. Make a pull request from shaloo/numpy.org branch issu92_sc_dom to NumPy/numpy.org 'master'
> >Go to GitHub in browser
> >Select issu92_sc_dom
> >Compare with NumPy/numpy.org master - make sure all my updates show up
> >Issue PR

All good.

> Once PR is merged, this is what I typically do - may be I need to change something in this workflow to avoid sync up with master conflicts.
> 
>     1. Sync up with upstream master
>        git fetch upstream
>        git checkout master
>        git merge upstream/master
>        git push origin master  (send updates to my fork master - is this the cause of merge issues that I am seeing?)

No it's not the cause of merge issue. There will be merge issues only if you have created new commits in your own `master`. If so, make sure that such commits are also present in another branch (so you don't lose them) and then do: 
```
git checkout master
git reset --hard upstream/master
```
Now your master branch is up to date with upstream master, and you should never get a merge conflict on master. 

> 
>     2. Validate that all issu92_sc_dom branch changes came back in resent with upstream master
>        Github browser - compare issu92_sc_dom from my fork with upstream master and if all changes made it delete my local branch using this command
>        git branch -d issu92_sc_dom
> 
> 
> At times, I see errors indicating all changes in issu92_sc_dom are not present in upstream master though a diff does not show any changes. For Data Science branch - i used a -D option and deleted it anyways .

Yep, this can happen if there was a squash merge. Using `-D` is correct.

Once a PR says "merged", you can be sure all changes in the PR are in master. You can check a few times in the beginning, but you should find that that's always the case.

>     1. Create a new branch corresponding to new issue/ticket and add content for a new tab / new page on new website.
> 
> 
> Repeat A-E and 1-2 for subsequent new feature/content addition.

Sounds good:)
